99-general-concepts/02-testing/02-unit-tests/README.md

```diff
@@ -0,0 +1,128 @@
+# Unit Testing
+
+- Tipo: `lectura`
+- Formato: `self-paced`
+- Duración: `30min`
+
+***
+
+## Objetivos de Aprendizaje
+
+- Understand how to write effective unit tests
+- Understand prinicples behind Test Driven Development
+
+***
+
+## Introduction
+
+By now you’ve noticed that at the end of every code challenge you’ve had to run tests against your code to verify that it works as expected. These are called unit tests. Each test programmatically ensures that your program works as intended by comparing actual outputs with expected values for static inputs.
+
+For example, consider a function that adds to numbers: `sum(a, b)`.
+
+To test `sum(a,b)`, we can select any two numbers and assert that the ouptut equals their expected sum:
+
+```javascript
+Assert.deepEqual(sum(2, 3), 5);
+```
+
+If the result is false, meaning `sum(2, 3)` equals something other than 5, we'll know we have some debugging to do!
+
+## Validating Credentials
+
+To explore unit tests further, let's analyize the `areValidCredentials` challenge:
+
+>Escribe una función llamada `areValidCredentials`.
+>
+>Dando un nombre y una contraseña, `areValidCredentials`, devuelve `true` si el nombre es superior a 3 caracteres, Y, la contraseña tiene al menos 8 caracteres de longitud. De lo contrario, devuelve `false`.
+
+Let's look at the unit tests associated with this challenge:
+
+```javascript
+describe("areValidCredentials()", function () {
+  it("debería regresar true cuando las credenciales son válidas", function () {
+	var output = areValidCredentials('Kemal', 'testtesttest');
+	Assert.deepEqual(output, true);
+  });
+
+  it("debería regresar false cuando el nombre es demasiado corto", function () {
+	var output = areValidCredentials('K', 'mylongpassword');
+	Assert.deepEqual(output, false);
+  });
+  
+  it("debería regresar false cuando la contraseña es demasiado corta", function () {
+	var output = areValidCredentials('Kemal', 'aah');
+	Assert.deepEqual(output, false);
+  });
+});
+```
+
+Each function here begins with the word “it”, which in English means “eso”. The “it” of each test refers to `areValidCredentials()` as declared in the wrapping `describe` function:
+
+`describe("areValidCredentials()", function(){...})`.
+
+You can see that each unit test is a function that reads like a sentence. For instance, this function...
+
+`it("debería regresar false cuando la contraseña es demasiado corta", function()) {....}`
```

Comment:
I think the curly braces (`{ ... }`) should be inside the `it()` and ends with a `;`